src/veins/base/toolbox/Signal.cc

```diff
@@ -555,268 +555,172 @@ Signal& Signal::operator+=(const Signal& other)
 {
     assert(this->getSpectrum() == other.getSpectrum());
 
+#ifndef PREFER_VECTORIZATION
     size_t lowIndex = std::min(this->getRelativeStart(), other.getRelativeStart());
     size_t highIndex = std::max(this->getRelativeEnd(), other.getRelativeEnd());
+#else
+    size_t lowIndex = 0;
+    size_t highIndex = numAbsoluteValues;
+#endif
 
-    for (size_t i = lowIndex; i < highIndex; i++) {
-        (*this)[i] += other[i];
-    }
+    std::transform(values + lowIndex, values + highIndex, other.values + lowIndex, values + lowIndex, std::plus<double>());
```

Comment:
Please add informative comment.
I agree that the std::transform-idiom makes a lot of sense here (e.g.,  to enable compiler optimization and parallelization). But it is harder to read than the simple loop. So let's balance that with a simple comment.
Same for many more instances below.

Reply:
You're right that `std::transform` is harder to parse than a simple use. I do think however, that this is mostly due to it's novelty, not because of its inherent complexity. Therefore I would prefer not to add a comment here, since this would basically only replicate the STL's documentation, as the use of `transform` here is a very basic one and has no pitfalls that would need to be explicitly described.